content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx

```diff
@@ -0,0 +1,258 @@
+
+</Admonition>
+
+### NoAction
+
+The `NoAction` action is similar to `Restrict` in that it will prevent the update or deletion of any records which are connected and throw an error. The difference between the two is dependant on the database being used. But in general they both have the same functionality. 
+
+In Postgres however, `NoAction` defers the check until later in the transaction.
+
+TODO::: add SQLSERVER, MYSQL
+
+#### Example usage
+
+```prisma file=schema.prisma highlight=4;add
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: NoAction)
+  authorId  Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `NoAction`?
+
+If a `User` has a related post in the `Post` table you will **not** be able to delete them. This will result in a runtime error.
+
+### SetNull
+
+The `SetNull` action will set relation [scalar fields](../data-model#scalar-fields) to null if the relation is updated or deleted. If one or more of the relations scalar fields are required a runtime error will be thrown.
+
+```prisma file=schema.prisma highlight=4;add
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: SetNull)
+  authorId  Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `SetNull`?
+
+If a `User` is deleted then the corresponding `authorId` field in the `Post` table will be set to `NULL`. It will be left in the database, floating there without a connection to any `User`.
+
+### SetDefault
+
+The `SetDefault` action will set relation [scalar fields](./#annotated-relation-fields-and-relation-scalar-fields) to their default values upon updated or deletion of a related record. If no defaults are provided for any of the scalar fields, a runtime error will be thrown.
+
+```prisma file=schema.prisma highlight=4;add
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: SetDefault)
+  authorId  Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `SetDefault`?
+
+If a `User` is deleted then the corresponding `authorId` field in the `Post` table will be set to its default value.
+
+## Behavior **without** the preview flag `referentialActions`
+
+<a id="current"></a>
+
+When invoking the [`delete()`](../../../../guides/database/advanced-database-tasks/cascading-deletes/postgresql#91-validating-restrict) or [`deleteAll()`](../../prisma-client/crud#delete-all-records) methods using the Prisma Client, a runtime check is performed and the deletion of records prevented if they are referencing related objects. **This prevents cascade behavior**. 
```

Comment:
Is this true for all relations or just some? (Question for the team that build this)

Reply:
@dpetrick ping